concordium-consensus/src/Concordium/GlobalState/TransactionTable.hs

```diff
@@ -260,16 +259,31 @@ emptyTransactionTableWithSequenceNumbers accs upds =
 -- | Add a transaction to a transaction table if its nonce/sequence number is at least the next
 --  non-finalized nonce/sequence number.  A return value of 'True' indicates that the transaction
 --  was added.  The caller should check that the transaction is not already present.
-addTransaction :: BlockItem -> CommitPoint -> TVer.VerificationResult -> TransactionTable -> (Bool, TransactionTable)
-addTransaction blockItem@WithMetadata{..} cp !verRes tt0 =
+addTransaction ::
+    -- | Transaction to add.
+    BlockItem ->
+    -- | Commit point of the transaction.
+    CommitPoint ->
+    -- | The associated verification result.
+    TVer.VerificationResult ->
+    -- | An optional "next account nonce" to initialize the @anftNextNonce@ with.
+    --  This MUST be provided for normal transactions.
+    Maybe Nonce ->
```

Comment:
There seems to be a lot of annoyance caused by this, but is this actually needed?

We seem to use it to
- double check our own work when finalizing transactions. If this fails the node crashes, so we should not need this in production.
- when adding transactions. But there we always do transaction preverification, which ensures that at least the nonce is >= nextNonce for the last finalized block. So the check we have here 

```
tt0 ^. senderANFT . anftNextNonce <= nonce 
```

should never be needed.